Created stubs for Infoblox plugin

renzon opened the pull request:
close #3811

rochacbruno commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,154 @@
+from robottelo.decorators import stubbed, tier3
+
+
+class InfobloxTestCase(unittest2.TestCase):
```
because of logging, cleanup and other things we do not direct inherit from unittest, please use `robottelo.test.TestCase` as base class.

rochacbruno commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,154 @@
+@Upstream: No
+"""
+
+import unittest2
```
`from robottelo.test import TestCase`

rochacbruno commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,154 @@
+
+@CaseLevel: Acceptance
+
+@CaseComponent: CLI
```
I guess  `@CaseCompoment`  must be other than **CLI** which stands for **hammer**, So maybe **@CaseComponent: INSTALLER** is the best option

rochacbruno commented:
ACK (pending comments)

oshtaier commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,153 @@
+
+        @Steps:
+
+        1. Pass appliance credentions via installer
```
`credentions` ?

renzon replied:
Have searched on Google, credentials is the correct word

omaciel replied:
@renzon right, but you spelled it `credenti**ons**` :)

oshtaier commented:
Use `CaseLevel: System` for all tier3 tests

oshtaier commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,153 @@
+    @tier3
+    def test_ip_range(self):
+        """Check Infoblox ip range is used when provisioning host is done while
+        Infloblox is settled as DHCP provider
```
`is set as DHCP provider`?

sghai replied:
Description can be defined as """Check host get IP from Infoblox IP range while provisioning a host""

oshtaier commented on tests/foreman/ui/test_capsule.py:
```diff
@@ -444,3 +446,95 @@ def test_positive_capsule_puppetca_autosign(self):
+
+        @id: 69acd8ec-79b8-11e6-8c44-68f72889dc7f
+
+
```
redundant empty line

oshtaier commented:
ACK pending comments

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,153 @@
+    @stubbed()
+    @tier3
+    def test_set_dhcp_provider(self):
+        """Check Infoblox DHCP plugin is settled as dhcp provider
```
`is set as`

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,153 @@
+        @Steps:
+
+        1. Set infoblox as dhcp provider with param
+        --foreman-proxy-dhcp-provider='infoblox'
```
Just a not we are still not sure about exact parameter names. Need to revisit this later.

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,153 @@
+
+        @Steps:
+
+        1. Enable Infoblox plugin via installer
```
I think we can define one liner step along with @Steps: tag ?
`@Steps: Enable Infoblox plugin via installer`

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,153 @@
+    @stubbed()
+    @tier3
+    def test_enable_dns_plugin(self):
+        """Check Infoblox DNS plugin can be enabled on host
```
s/host/server ?

sghai commented on tests/foreman/ui/test_capsule.py:
```diff
@@ -444,3 +446,95 @@ def test_positive_capsule_puppetca_autosign(self):
+    @stubbed()
+    @tier3
+    def test_enable_infoblox_dns_plugin(self):
+        """Check Infoblox DNS plugin can be enabled on capsule
```
` tests/foreman/ui/test_capsule.py` contains all the tests related to UI  capsule  page. 

So either we can rename this test like:
"""Check infoblox DNS feature is enabled on capsule"""
and assert: DNS should be listed under capsule features

OR

Add new test as suggested above and move all installer related tests where scenario is capsule to `tests/foreman/installer/test_infoblox.py`

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,221 @@
+
+        @Steps: Set infoblox as dhcp provider with param
+        --foreman-proxy-dhcp-provider='infoblox'.
+         Obs: the exactly param is not yest defined
```
I think we can remove this line now `Obs: the exactly param is not yest defined` as we may need to revisit all stubs again with exact param name.

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,221 @@
+        @Assert: Check host dns is updated accordingly to infoblox
+
+        @caseautomation: notautomated
+        """
```
`@CaseLevel: System` needs to be updated here

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,221 @@
+        @Assert: Check host ip is on infoblox range
+
+        @caseautomation: notautomated
+        """
```
@CaseLevel: System needs to be updated here

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,221 @@
+        @Assert: Check inflobox is set as DHCP provider
+
+        @caseautomation: notautomated
+        """
```
@CaseLevel: System needs to be updated here

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,221 @@
+        @Assert: Check DHCP plugin is disabled on host
+
+        @caseautomation: notautomated
+        """
```
@CaseLevel: System needs to be updated here

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,221 @@
+        infoblox_hostname, username & password
+
+        @caseautomation: notautomated
+        """
```
@CaseLevel: System needs to be updated here

sghai commented on tests/foreman/installer/test_infoblox.py:
```diff
@@ -0,0 +1,221 @@
+        @Assert: Check DNS plugin is enabled on host
+
+        @caseautomation: notautomated
+        """
```
@CaseLevel: System needs to be updated here

sghai commented:
@renzon : Are you done with changes ? or still something needs to be updated here ?

renzon commented:
@sghai I am done. I have changed everything asked and added params based on [community video](https://www.youtube.com/watch?v=GFNPHypFhl4&feature=youtu.be&t=1350) you sent me.

sghai commented:
ACK
